src/web3-utils.js

```diff
@@ -48,32 +89,35 @@ export function formatBalance(
   return `${whole}.${fraction}`
 }
 
-/**
- * Shorten an Ethereum address. `charsLength` allows to change the number of
- * characters on both sides of the ellipsis.
- *
- * Examples:
- *   shortenAddress('0x19731977931271')    // 0x1973…1271
- *   shortenAddress('0x19731977931271', 2) // 0x19…71
- *   shortenAddress('0x197319')            // 0x197319 (already short enough)
- *
- * @param {string} address The address to shorten
- * @param {number} [charsLength=4] The number of characters to change on both sides of the ellipsis
- * @returns {string} The shortened address
+export function getEmptyAddress() {
+  return EMPTY_ADDRESS
+}
+
+/*
+ * Return the injected provider, if any.
  */
-export function shortenAddress(address, charsLength = 4) {
-  const prefixLength = 2 // "0x"
-  if (!address) {
-    return ''
+export function getInjectedProvider() {
+  if (window.ethereum) {
+    return window.ethereum
   }
-  if (address.length < charsLength * 2 + prefixLength) {
-    return address
+  if (window.web3 && window.web3.currentProvider) {
+    return window.web3.currentProvider
+  }
+  return null
+}
+
+// Get the first account of a web3 instance
+export async function getMainAccount(web3) {
+  try {
+    const accounts = await web3.eth.getAccounts()
+    return (accounts && accounts[0]) || null
```

Comment:
I remember I was getting a blank (`''`) account, is that a valid thing? If not we may need to update this check, if it is then you can disregard this comment.

Reply:
Which ethereum provider were you using? Was it an empty string or `undefined`? It'd be really weird if one returned `['']` for `getAccounts()` 🤷‍♂️

Reply:
`Coinbase`, `Cypher` and `Status`: I found this during my quest to figure out what was going on in the mobile providers (before we introduced `getInjectedProvider`)

Reply:
Hmm, that definitely seems like a weird thing to happen on `getAccounts()`; let's monitor and see if this still happens 👍